extensions/azdata/src/extension.ts

```diff
@@ -10,26 +10,30 @@ import { ExitCodeError } from './common/childProcess';
 
 export async function activate(): Promise<void> {
 	const outputChannel = vscode.window.createOutputChannel('azdata');
-	if (false) {
-		await checkForAzdata(outputChannel);
-	}
+	await checkForAzdata(outputChannel);
 }
 
 async function checkForAzdata(outputChannel: vscode.OutputChannel): Promise<void> {
 	try {
 		const azdata = await findAzdata(outputChannel);
 		vscode.window.showInformationMessage(loc.foundExistingAzdata(azdata.path, azdata.version));
 	} catch (err) {
-		const response = await vscode.window.showErrorMessage(loc.couldNotFindAzdataWithPrompt, loc.install, loc.cancel);
-		if (response === loc.install) {
-			try {
-				await downloadAndInstallAzdata(outputChannel);
-				vscode.window.showInformationMessage(loc.azdataInstalled);
-			} catch (err) {
-				// Windows: 1602 is User Cancelling installation - not unexpected so don't display
-				if (!(err instanceof ExitCodeError) || err.code !== 1602) {
-					vscode.window.showWarningMessage(loc.installError(err));
-				}
+		// Don't block on this since we want the extension to finish activating without needing user input.
+		// Calls will be made to handle azdata not being installed
+		promptToInstallAzdata(outputChannel).catch(e => console.log(`Unexpected error prompting to install azdata ${e}`));
```

Comment:
Should the unexpected error encountered not go to outputChannel so that it is easier to debug if and when that happens?

Reply:
Will revisit when I go over logging. But this one is for truly unexpected errors that I wouldn't expect the user to be able to do anything about.